background.ts

```diff
@@ -504,8 +503,29 @@ ipcMainProxy.on('preferences-set-dirty', (_event, dirtyFlag) => {
 });
 
 function writeSettings(arg: RecursivePartial<RecursiveReadonly<settings.Settings>>) {
-  // arrayCustomizer is necessary to properly merge array of strings
-  _.mergeWith(cfg, arg, arrayCustomizer);
+  const customizer = (objValue: any, srcValue: any) => {
```

Comment:
Thinking that if this whole block got moved into `utils/object.js` or a newer function we can unit-test it more easily. I get what it's doing, but think it would be worth having unit tests for it because there are so many edge cases.